core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1686,12 +1847,208 @@ public void warmUp() {
+                    if (ColumnType.isVariableLength(columnType)) {
+                        removeFileAndOrLog(ff, iFile(other2.parent(), columnName, columnNameTxn));
+                    } else if (ColumnType.isSymbol(columnType)) {
+                        removeFileAndOrLog(ff, BitmapIndexUtils.keyFileName(other2.parent(), columnName, columnNameTxn));
+                        removeFileAndOrLog(ff, BitmapIndexUtils.valueFileName(other2.parent(), columnName, columnNameTxn));
+                    }
+                }
+
+                // delete extra index files
+                int other2Len = other2.parent().length();
+                keys = detachedRemoveIndexColNames.keys();
+                for (int i = 0, limit = keys.size(); i < limit; i++) {
+                    CharSequence columnName = keys.get(i);
+                    int colIdx = detachedRemoveIndexColNames.get(columnName);
+                    long columnNameTxn = detachedColumnVersionReader.getColumnNameTxn(partitionTimestamp, colIdx);
+                    removeFileAndOrLog(ff, BitmapIndexUtils.keyFileName(other2.trimTo(other2Len), columnName, columnNameTxn));
+                    removeFileAndOrLog(ff, BitmapIndexUtils.valueFileName(other2.trimTo(other2Len), columnName, columnNameTxn));
+                }
+
+                // reindex
+                int limit = detachedReIndexColNames.size();
+                if (limit > 0) {
+                    if (detachedIndexBuilder == null) {
+                        detachedIndexBuilder = new IndexBuilder();
+                    }
+                    PartitionBy.setSinkForPartition(other.trimTo(0), partitionBy, partitionTimestamp, false);
+                    other.put(ATTACHABLE_DIR_MARKER);
+                    detachedIndexBuilder.of(other2.trimTo(other2Len).parent(), configuration);
+                    for (int i = 0; i < limit; i++) {
+                        CharSequence columnName = detachedReIndexColNames.get(i);
+                        detachedIndexBuilder.reindexColumn(
+                                detachedColumnVersionReader,
+                                metadata,
+                                metadata.getColumnIndex(columnName),
+                                other,
+                                -1L,
+                                partitionTimestamp,
+                                partitionSize
+                        );
+                    }
+                }
+                deleteDetachedMetadataFiles = true;
+            }
+        } finally {
+            Misc.free(detachedColumnVersionReader);
+            if (detachedMetaMem != null) {
+                detachedMetaMem.close();
+            }
+            if (detachedIndexBuilder != null) {
+                detachedIndexBuilder.clear();
+            }
+            if (deleteDetachedMetadataFiles) {
+                removeFileAndOrLog(ff, other2.of(detachedPath).concat(DETACHED_META_FILE_NAME).$());
```

Comment:
please play safe, do 

```
removeFileAndOrLog(ff, detachedPath.of(configuration.getDetachedRoot()).concat(tableName).concat(DETACHED_META_FILE_NAME).$());
```

here and second line

Reply:
You also want to do those remove at the end of the try, not in the finally block.